pkg/prometheus/resource_selector.go

```diff
@@ -1035,6 +1035,14 @@ func (rs *ResourceSelector) validateConsulSDConfigs(ctx context.Context, sc *mon
 			return fmt.Errorf("field `config.Namespace` is only supported for Prometheus version >= 2.28.0")
 		}
 
+		if config.Filter != nil && rs.version.Major < 3 {
+			return fmt.Errorf("field `config.Filter` is only supported for Prometheus version >= 3.0.0")
+		}
+
+		if config.Filter != nil && (len(config.NodeMeta) > 0 || len(config.Tags) > 0) {
+			rs.l.Warn("as of Consul version >= 1.14, the `config.Tags` and `config.NodeMeta` fields are deprecated in Consul in favor of `config.Filter`")
```

Comment:
nit: Here also since this is consul specific deprecation should we log it? I have no strong opinion but not sure
@simonpasquier wdyt?

Reply:
The general idea here is to log deprecated fields in the CRDs. Since the deprecation is at the Consul level, I wouldn't consider it necessary to log here.

Reply:
Thanks @simonpasquier @slashpai 
I got it
Remove it done.